rangeproof/proof.go

```diff
@@ -435,6 +439,27 @@ func (p *Proof) Proves(statement *Statement) bool {
 	return p.ProvesStatement(statement.Sign, statement.Factor, statement.Bound)
 }
 
+// ProvenStatement returns the statement that this proof proves. Calling the second and third return
+// parameters "factor" and "bound" respectively, then
+//    factor*attribute - bound >= 0  or  <= 0
+// where the inequality type is returned as the first parameter.
+//
+// NB: this method does not verify the proof. Do not trust the output unless proof.Verify() has been
+// invoked first.
+func (p *Proof) ProvenStatement() (StatementType, uint, *big.Int) {
+	bound := new(big.Int).Set(p.K)
+	factor := p.A
+	if len(p.Cs) == 3 {
+		bound.Add(bound, big.NewInt(2)).Rsh(bound, 2)
+		factor >>= 2
+	}
+	typ := GreaterOrEqual
```

Comment:
in the `Sign()` function you use a switch statement. i'd use a similar one here for readability